src/sync.coffee

```diff
@@ -83,6 +83,11 @@ module.exports = (method, model, options) ->
   if params.type == 'PATCH' && window.ActiveXObject && !(window.external && window.external.msActiveXFilteringEnabled)
     params.xhr = -> new ActiveXObject('Microsoft.XMLHTTP')
 
+  beforeSend = options.beforeSend
+  options.beforeSend = (xhr) ->
+    xhr.setRequestHeader 'X-Feature-Name', options.featureName if options.featureName
+    beforeSend.apply this, arguments if beforeSend
```

Comment:
CoffeeScript magic gives us

`beforeSend?.apply this, arguments`

Reply:
👍 I prefer that